ckanext/canada/validators.py

```diff
@@ -531,3 +531,35 @@ def protect_registry_access(key, data, errors, context):
                            " from '%s' to '%s'. This field is read-only." %
                            (original, value)))
         raise StopOnError
+
+
+def limit_resources_per_dataset(key, data, errors, context):
+    """
+    Limits the number of resources per dataset.
+    """
+    package_id = data.get(key[:-1] + ('id',))
+
+    max_resource_count = config.get('ckanext.canada.max_resources_per_dataset', None)
+    if not max_resource_count:
+        return
+
+    new_resource_count = len(set(k[1] for k in data.keys() if k[0] == 'resources'))
+    if not new_resource_count:
+        return
+
+    current_resource_count = model.Session.query(model.Resource.id)\
+        .filter(model.Resource.package_id == package_id)\
+        .filter(model.Resource.state == 'active').count()
+    if not current_resource_count:
+        return
```

Comment:
I'd move this into the `if new_resource_count > int(max_resource_count):` block so you can skip the extra db query most of the time